Uncaught exception print in stderr when timeout

**Describe the bug**

When a timeout occurs, the behavior is correct but, sometimes, a race condition happens and an unhandled exception is thrown.

**To Reproduce**

```
public class TimeoutTest {

    public static void main(String[] args) {
        ProcBuilder builder = new ProcBuilder("bash",  "-i", "-c", "echo hello").withTimeoutMillis(1);

        try {
            builder.run();
        } catch (TimeoutException ex) {
            //OK
        }
    }
}
```

**Expected behavior**

No uncaught exception should be printed in stderr.

**Actual Behaviour**

These traces are printed by stderr

```
Exception in thread "Thread-1" java.lang.RuntimeException: ${END}
	at org.buildobjects.process.Proc.dispatch(Proc.java:168)
	at org.buildobjects.process.ByteArrayConsumptionThread$1.run(ByteArrayConsumptionThread.java:36)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.InterruptedException
	at java.base/java.util.concurrent.locks.ReentrantLock$Sync.lockInterruptibly(ReentrantLock.java:159)
	at java.base/java.util.concurrent.locks.ReentrantLock.lockInterruptibly(ReentrantLock.java:372)
	at java.base/java.util.concurrent.LinkedBlockingQueue.put(LinkedBlockingQueue.java:332)
	at org.buildobjects.process.Proc.dispatch(Proc.java:166)
	... 2 more
Exception in thread "Thread-0" java.lang.RuntimeException: ${END}
	at org.buildobjects.process.Proc.dispatch(Proc.java:168)
	at org.buildobjects.process.StreamCopyConsumptionThread$1.run(StreamCopyConsumptionThread.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.InterruptedException
	at java.base/java.util.concurrent.locks.ReentrantLock$Sync.lockInterruptibly(ReentrantLock.java:159)
	at java.base/java.util.concurrent.locks.ReentrantLock.lockInterruptibly(ReentrantLock.java:372)
	at java.base/java.util.concurrent.LinkedBlockingQueue.put(LinkedBlockingQueue.java:332)
	at org.buildobjects.process.Proc.dispatch(Proc.java:166)
	... 2 more
```

I didn't expect any uncaught exception.

**Context**

I did use Ubuntu.

--- a/src/main/java/org/buildobjects/process/ByteArrayConsumptionThread.java
+++ b/src/main/java/org/buildobjects/process/ByteArrayConsumptionThread.java
@@ -32,8 +32,10 @@ class ByteArrayConsumptionThread implements OutputConsumptionThread {
                 try {
                     bytes = toByteArray(inputStream);
                 } catch (Throwable t) {
-                    ByteArrayConsumptionThread.this.throwable = t;
-                    eventSink.dispatch(EXCEPTION_IN_STREAM_HANDLING);
+                    if (!thread.isInterrupted()) {
+                        ByteArrayConsumptionThread.this.throwable = t;
+                        eventSink.dispatch(EXCEPTION_IN_STREAM_HANDLING);
+                    }
                 }
             }
         });

--- a/src/main/java/org/buildobjects/process/StreamConsumerConsumptionThread.java
+++ b/src/main/java/org/buildobjects/process/StreamConsumerConsumptionThread.java
@@ -28,8 +28,10 @@ class StreamConsumerConsumptionThread implements OutputConsumptionThread {
                     stdout.consume(inputStream);
 
                 } catch (Throwable t) {
-                    StreamConsumerConsumptionThread.this.throwable = t;
-                    eventSink.dispatch(EXCEPTION_IN_STREAM_HANDLING);
+                    if (!thread.isInterrupted()) {
+                        StreamConsumerConsumptionThread.this.throwable = t;
+                        eventSink.dispatch(EXCEPTION_IN_STREAM_HANDLING);
+                    }
                 }
             }
         });

--- a/src/main/java/org/buildobjects/process/StreamCopyConsumptionThread.java
+++ b/src/main/java/org/buildobjects/process/StreamCopyConsumptionThread.java
@@ -25,11 +25,14 @@ class StreamCopyConsumptionThread implements OutputConsumptionThread {
             public void run() {
                 try {
                     new StreamCopyRunner(inputStream, stdout, false).run();
-            } catch (Throwable t) {
-                StreamCopyConsumptionThread.this.throwable = t;
-                eventSink.dispatch(EXCEPTION_IN_STREAM_HANDLING);
+                } catch (Throwable t) {
+                    if (!thread.isInterrupted()) {
+                        StreamCopyConsumptionThread.this.throwable = t;
+                        eventSink.dispatch(EXCEPTION_IN_STREAM_HANDLING);
+                    }
+                }
             }
-        }});
+        });
         this.thread.start();
     }
 

